presto-spark-base/src/main/java/com/facebook/presto/spark/PrestoSparkConfig.java

```diff
@@ -477,4 +499,38 @@ public PrestoSparkConfig setAdaptiveJoinSideSwitchingEnabled(boolean adaptiveJoi
         this.adaptiveJoinSideSwitchingEnabled = adaptiveJoinSideSwitchingEnabled;
         return this;
     }
+
+    /**
+     * Parses the catalog properties from the input string in format catalog.key=value and returns a nested map of properties.
+     *
+     * @throws IllegalArgumentException If the input string or any of its properties are not correctly formatted.
+     */
+    public static Map<String, Map<String, String>> parseCatalogProperties(String input)
+            throws IllegalArgumentException
+    {
+        try {
+            Map<String, String> inputMap = MAP_SPLITTER.split(nullToEmpty(input));
+
+            Map<String, Map<String, String>> result = new HashMap<>();
+
+            for (Map.Entry<String, String> entry : inputMap.entrySet()) {
+                String key = entry.getKey();
+                String value = entry.getValue();
+
+                String[] parts = key.split("\\.");
+                if (parts.length != 2) {
+                    throw new IllegalArgumentException("Property " + key + " is not correctly formatted");
```

Comment:
you throw an exception here, but then catch it at the end and rethrow a different one.